### PR TITLE
refactor: toast handling

### LINKED ISSUE #279: [appStore] refactor showToast signature
Instead of a fixed list of arguments, it would be more straightforward to pass in a `Toast` object.

### PR DESCRIPTION
**Pull request by rofe**
Fix #279

### CODE REVIEW

File: src/extension/app/components/action-bar/activity-action/activity-action.js [hunk 1]
```diff
@@ -45,10 +45,6 @@ export class ActivityAction extends ConnectedElement {
   }
 
   handleCloseToast() {
-    if (this.appStore.toast.closeCallback) {
-      this.appStore.toast.closeCallback();
-      return;
-    }
```
Comment (rofe) [inline src/extension/app/components/action-bar/activity-action/activity-action.js:51]: Moved to `AppStore.closeToast()`

File: src/extension/types/typedefs.js [hunk 2]
```diff
@@ -185,14 +185,14 @@
...
+ * @prop {string} [variant] The toast variant
  * @prop {Function} [closeCallback] The function to call on close
  * @prop {Function} [actionCallback] The function to call on action
  * @prop {string} [actionLabel] The action label
  * @prop {Function} [secondaryCallback] The function to call on secondary action
  * @prop {string} [secondaryLabel] The secondary actiom label
- * @prop {number} [timeout] The time the toast is shown (default: 6000)
- * @prop {boolean} [actionOnTimeout=true] Execute action on timeout
+ * @prop {number} [timeout] The time the toast is shown (default: 3000)
+ * @prop {Function} [timeoutCallback] The function to call on timeout
```
Comment (rofe) [inline src/extension/types/typedefs.js:195]: Added a dedicated callback for timeouts

File: test/app/plugins/delete/delete.test.js [hunk 3]
```diff
@@ -65,196 +65,202 @@ function confirmDelete(sidekick) {
...
+      confirmDelete(sidekick);
+
+      await waitUntil(() => deleteStub.calledOnce);
+      expect(deleteStub.calledOnce);
+    });
+  });
+
+  await runTests(HelixMockContentType.DOC);
+  await aTimeout(5000);
+  await runTests(HelixMockContentType.SHEET);
```
Comment (rofe) [inline test/app/plugins/delete/delete.test.js:265]: Refactored tests to prevent concurrency issues while waiting for the toast timeouts.

**Review by dylandepass — APPROVED** [review]

**Pull request merged**